src/core/get-sheet.js

```diff
@@ -12,13 +12,16 @@ export let getSheet = (target) => {
     if (typeof window === 'object') {
         // Querying the existing target for a previously defined <style> tag
         // We're doing a querySelector because the <head> element doesn't implemented the getElementById api
-        return (
+        const styleElem =
             (target ? target.querySelector('#' + GOOBER_ID) : window[GOOBER_ID]) ||
             Object.assign((target || document.head).appendChild(document.createElement('style')), {
                 innerHTML: ' ',
                 id: GOOBER_ID
-            })
-        ).firstChild;
+            });
+        // if style el is empty populate it with a comment, else it will return `null` and fail out if
+        // anyone attempts to access it's `data` property
+        if (!styleElem.firstChild) styleElem.innerHTML = '/*goober*/';
```

Comment:
Thank you @anied for this! I've seen folks get annoyed with this so this is truly needed.

Should we do the first element check only for the existing tags? Meaning the assign call adds the missing tag with the empty space. Which brings me to the next question, instead of a comment could we use an empty space?

Reply:
Hello @cristianbote !

> Should we do the first element check only for the existing tags? Meaning the assign call adds the missing tag with the empty space.

I'm not sure I completely follow this question. It _is_ true that the `if (!styleElem.firstChild) styleElem.innerHTML = '/*goober*/';` line is superfluous for cases in which this is how the style block is added:

```
Object.assign((target || document.head).appendChild(document.createElement('style')), {
  innerHTML: ' ',
  id: GOOBER_ID
});
``` 

However, I knew that there was a big focus on keeping the size of the library down, so I figured a potential superfluous check would be preferable to a larger bundle; I'm happy to write this to avoid the superfluous call if that's desired.  Am I understanding the ask correctly?

> Which brings me to the next question, instead of a comment could we use an empty space?

Yes, I believe this should be totally fine, but I could verify it later.

Reply:
Ah sorry for not giving a more deeper explanation. I'll do my best to add more details.

```js
const currentStyleTag = target ? target.querySelector('#' + GOOBER_ID) : window[GOOBER_ID];

if (currentStyleTag) {
  if (!currentStyleTag.firstChild) {
    currentStyleTag.innerHTML = ' ';
  }

  return currentStyleTag.firstChild;
}

return Object.assign((target || document.head).appendChild(document.createElement('style')), {
  innerHTML: ' ',
  id: GOOBER_ID
}).firstChild;
```

Not sure how much size this would add though 😅

Reply:
Hahaha, OK, yes-- this was my first instinct, but seeing the push for keeping the size down was the reason I went with the approach I did.  I'm happy to rework it to be a bit more readable and efficient at the expense of a few more bytes if you thinks that's acceptable.

Reply:
I'm not at my desk now but frankly I think what we have here is what's gonna get generated at the end as well.

Reply:
OK, cool-- I can refactor to this approach in a few hours.  Thanks!